modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/learn_message/LearnMessage.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLink from '@clayui/link';
+import getCN from 'classnames';
+import React, {useContext} from 'react';
+
+export const LearnResourcesContext = React.createContext({});
+
+interface IProps {
+	className?: string;
+
+	/**
+	 * The learn resource
+	 */
+	resource: string;
+
+	/**
+	 * The key to render.
+	 */
+	resourceKey: string;
+}
+
+/**
+ * This component is used to render links to Liferay Learn articles. The json
+ * object `learnResources` contains the messages and urls and is taken from
+ * liferay-portal/learn-resources.
+ *
+ * Use `LearnResourcesContext` to wrap the entire React App.
+ *
+ * Example use:
+ * <LearnResourcesContext.Provider value={learnResources}>
+ * 	<LearnMessage resourceKey="general" resource="portlet-configuration-web" />
+ * </LearnResourcesContext.Provider>
+ *
+ * Example of `learnResources`:
+ * {
+ * 	"portlet-configuration-web": { // Learn resource
+ *		"general": { // Resource key
+ *			"en_US": {
+ *				"message": "Tell me more",
+ *				"url": "https://learn.liferay.com/"
+ *			}
+ *		}
+ * 	}
+ * }
+ */
+export default function LearnMessage({
+	className = '',
+	resource,
+	resourceKey,
+	...otherProps
```

Comment:
I think this is irrelevant here because we don't declare any other props in the IProps interface.  One option would be to extend the interface from ClayLink props

Reply:
Extending the ClayLink interface seems appropriate but it's not currently exported, so instead I've added `extends React.AnchorHTMLAttributes<HTMLAnchorElement>` for the moment.

Reply:
I think you can do something like this

```js
type ClayLinkProps = React.ComponentProps<typeof ClayLink>

// You may also be able to do this

type ClayLinkProps = ClayLink['props']
```